File: EcommerceDDD.Application/Carts/SaveCart/SaveCartRequest.cs
using System;
using EcommerceDDD.Application.Carts;

namespace EcommerceDDD.Application.Carts.CreateCart
{
    public class SaveCartRequest
    {
        public Guid CustomerId { get; set; }
        public ProductDto Product { get; set; }
        public string Currency { get; set; }
    }
}

File: src/EcommerceDDD.Spa/src/app/core/models/requests/SaveCartRequest.ts
import { Product } from '../Product';

export class SaveCartRequest {
  customerId: string;
  product: Product;
  currency: string;

  constructor(customerId: string, product: Product, currency: string) {
    this.customerId = customerId;
    this.product = product;
    this.currency = currency;
  }
}
